doc/sailfish-webview-qmozcontext.cpp

```diff
@@ -0,0 +1,367 @@
+    be created). By default this is set to
+    \c QStandardPaths::writableLocation(QStandardPaths::CacheLocation), which
+    is usually \c{~/.cache/<organisation name>/<application name>}.
+
+    Generally there is no need to call \c setProfile as the profile will
+    automatically set up on initialization, and can also be passed as
+    parameter to \l{WebEngine::initialize}.
+
+    When WebEngine is instantiated as a QML component this is also called
+    automatically during initalisation. However, it can be called earlier to
+    set a custom profile location in case the intialization process is being
+    overridden.
+
+    \note If using a different location care is needed to select a path that's
+    acceptable from an application sandboxing point of view. The qstandard
+    application paths are safe for profile paths.
+
+    See \l {SailfishOS::WebEngine::initialize}{WebEngine::initialize} for more
+    info.
+
+    \sa {SailfishOS::WebEngine::initialize}{WebEngine::initialize}, runEmbedding, stopEmbedding
+*/
+void QMozContext::setProfile(const QString &profilePath);
+
+/*!
+    \internal
+    \brief Called automatically on WebView creation so that the engine can create views.
+
+    For internal use.
+
+    \sa setViewCreator
+*/
+void QMozContext::setViewCreator(QMozViewCreator *viewCreator);
+
+/*!
+    \internal
+    \brief Opens a new view that can contain webcontent.
+
+    For internal use.
+
+    \sa setViewCreator
+*/
+void QMozContext::createView();
+
+/*!
+    \brief This signal is emitted when a notification on an observed topic is received.
+
+    After an app has registered an interest in a particular topic by calling
+    \l addObserver or \l addObservers, it can then conect to this signal to be
+    notified about any component sending a notification on the topic.
+
+    The \a message parameter will be the topic in question, the \a data parameter
+    will contain data sent by the sender constructed from a JSON object. The
+    exact object structure is at the sender's discretion.
+
+    \sa addObserver, addObservers, removeObserver, removeObservers, notifyObservers
+*/
+void QMozContext::recvObserve(const QString message, const QVariant data);
+
+/*!
+    \brief This signal is emitted when the gecko EmbedLiteApp has been destroyed.
+
+    This signal is emitted when the EmbedLiteApp has been destroyed, just prior
+    to the WebEngine being deleted and the program exiting.
+*/
+void QMozContext::contextDestroyed();
+
+/*!
+    \brief This signal is emitted when when the last view has been destroyed.
+*/
+void QMozContext::lastViewDestroyed();
+
+/*!
+    \brief This signal is sent when the last registered window has been destroyed.
+
+    By default this is used to delay execution of \l stopEmbedding until after
+    the last window has closed.
+*/
+void QMozContext::lastWindowDestroyed();
+
+/*!
+    \brief Regsiter javascript chrome components to be loaded into the WebEngine.
```

Comment:
Regsiter -> Register

Reply:
Also, maybe we should use camel-casing for javascript -> JavaScript ?

Reply:
Thanks. I've updated all of the JavaScripts too.